tcp-server/src/main.rs

```diff
@@ -50,47 +45,53 @@ impl VecDequeMailbox {
 
 /// General questions:
 /// * how to work with `dyn Mailbox ` in multithreaded environment?
-fn main() -> io::Result<()> {
-    spawn_tcp_listener();
-    let client_thread = spawn_monitoring_thread();
-    client_thread.join().unwrap();
+#[tokio::main]
+async fn main() -> io::Result<()> {
+    tokio::join!(spawn_monitoring_thread(), spawn_tcp_listener(),);
     Ok(())
 }
 
 /// Spawns TcpListener thread, connects to 127.0.0.1:8080 and spawns thread per connection
-fn spawn_tcp_listener() {
-    thread::spawn(|| {
+fn spawn_tcp_listener() -> JoinHandle<()> {
```

Comment:
Nice way of doing it - a seperate tokio task which spawns more tasks. This resolves an issue I had. Thanks